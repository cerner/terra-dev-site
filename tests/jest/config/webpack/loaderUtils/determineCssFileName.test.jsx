import findCss from '../../../../../config/webpack/loaderUtils/determineCssFileName';

describe('devSiteExample css finder', () => {
  it('separates a shortened css file path correctly', () => {
    const testText = 'Import from \'./pretendExampleCss.scss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('separates a direct css file correctly', () => {
    const testText = 'Import from \'/Users/user1/terra-dev-site/tests/jest/config/webpack/loaders/pretendExampleCss.scss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('returns the whole file name when the import contains css and scss', () => {
    const testText = 'Import from \'./css.module.scss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('returns the first import when there are multiple', () => {
    const testText = 'Import from \'./pretendExample1.scss\'\nImport from \'./pretendExample2.scss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('returns undefined if there is no .css or .scss file declaration', () => {
    const testText = 'Import from \'./pretendExample\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('returns undefined if there is not file declaration but the file name contains css', () => {
    const testText = 'Import from \'./pretendExampleCss\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });

  it('returns undefined when the import is commented out', () => {
    const testText = '// Import from \'./pretendExampleCss.css\'';
    const example = findCss(testText);
    expect(example).toMatchSnapshot();
  });
});
