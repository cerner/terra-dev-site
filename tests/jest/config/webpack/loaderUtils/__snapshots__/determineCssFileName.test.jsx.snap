// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`devSiteExample css finder returns the first import when there are multiple 1`] = `"./pretendExample1.scss"`;

exports[`devSiteExample css finder returns the whole file name when the import contains css and scss 1`] = `"./css.module.scss"`;

exports[`devSiteExample css finder returns undefined if there is no .css or .scss file declaration 1`] = `undefined`;

exports[`devSiteExample css finder returns undefined if there is not file declaration but the file name contains css 1`] = `undefined`;

exports[`devSiteExample css finder returns undefined when the import is commented out 1`] = `undefined`;

exports[`devSiteExample css finder separates a direct css file correctly 1`] = `"/Users/user1/terra-dev-site/tests/jest/config/webpack/loaders/pretendExampleCss.scss"`;

exports[`devSiteExample css finder separates a shortened css file path correctly 1`] = `"./pretendExampleCss.scss"`;
